pkg/network/protocols/postgres/model_linux.go

```diff
@@ -60,29 +61,30 @@ func (e *EventWrapper) Operation() Operation {
 	return e.operation
 }
 
-var (
-	regexMapper = map[Operation]*regexp.Regexp{
-		CreateTableOP: regexp.MustCompile(`CREATE TABLE (\S+)`),
-		InsertOP:      regexp.MustCompile(`INSERT INTO (\S+)`),
-		DropTableOP:   regexp.MustCompile(`DROP TABLE(\s*IF\s+EXISTS\s+)?(\S+)`),
-		UpdateOP:      regexp.MustCompile(`UPDATE (\S+)`),
-		SelectOP:      regexp.MustCompile(`SELECT .* FROM (\S+)`),
-	}
-)
-
 // extractTableName extracts the table name from the query.
 func (e *EventWrapper) extractTableName() string {
-	regex, ok := regexMapper[e.Operation()]
-	if ok {
-		if matches := regex.FindSubmatch(e.Tx.getFragment()); len(matches) > 1 {
-			return string(
-				bytes.Trim( // Remove leading and trailing quotes from the table name
-					bytes.ReplaceAll(matches[len(matches)-1], []byte{0}, []byte{}), // Remove null bytes
-					"\""),
-			)
-		}
+	fragment := string(e.Tx.getFragment())
+	// Check if the string contains "IF EXISTS",
+	// temp solution for the fact that ObfuscateSQLString does not support "IF EXISTS".
+	if strings.Contains(fragment, "IF EXISTS") {
+		fragment = strings.ReplaceAll(fragment, "IF EXISTS", "")
+	}
+
+	normalizer := sqllexer.NewNormalizer(
+		sqllexer.WithCollectTables(true),
+	)
+
+	// Normalize the query without obfuscating it.
+	_, statementMetadata, err := normalizer.Normalize(fragment, sqllexer.WithDBMS("postgresql"))
+	if err != nil {
+		log.Warnf("unable to normalize due to: %s", err)
+		return "UNKNOWN"
 	}
-	return "UNKNOWN"
+	if statementMetadata.Size == 0 {
+		return "UNKNOWN"
+	}
+	return strings.Join(statementMetadata.Tables, ",")
```

Comment:
why joining? 
please create a UT to demonstrate the behvaior

Reply:
The idea is to able to support also queries which are more complex, and contains multiple table names. Added a UT to show the use case.

Reply:
commented in the test case